src/main/java/com/ericsson/ei/subscriptionhandler/SubscriptionValidator.java

```diff
@@ -16,153 +16,188 @@
+		if (subscription.getNotificationType().equals("REST_POST")) {
+			this.RestPostMediaType(subscription.getRestPostBodyMediaType());
+		}
+		LOGGER.debug("Validating of subscription " + subscription.getSubscriptionName() + " finished successfully.");
+	}
+
+	/**
+	 * Validation of subscriptionName parameter Throws
+	 * SubscriptionValidationException if validation of the parameter fails due to
+	 * wrong format of parameter.
+	 *
+	 * @param subscriptionName
+	 */
+	private void validateSubscriptionName(String subscriptionName) throws SubscriptionValidationException {
+		String regex = "^[A-Za-z0-9_]+$";
+		if (subscriptionName == null) {
+			throw new SubscriptionValidationException("Required field SubscriptionName has not been set");
+		} else if (!Pattern.matches(regex, subscriptionName)) {
+			throw new SubscriptionValidationException("Wrong format of SubscriptionName: " + subscriptionName);
+		}
+	}
+
+	/**
+	 * Validation of NotificationMessageKeyValues parameters (key/values) Throws
+	 * SubscriptionValidationException if validation of the parameter fails due to
+	 * wrong format of parameter.
+	 *
+	 * @param notificationMessage
+	 * @param restPostBodyMediaType
+	 */
+
+	private void validateNotificationMessageKeyValues(List<NotificationMessageKeyValue> notificationMessage,
+			String restPostBodyMediaType) throws SubscriptionValidationException {
+		for (NotificationMessageKeyValue item : notificationMessage) {
+			String testKey = item.getFormkey();
+			String testValue = item.getFormvalue();
+			if (restPostBodyMediaType != null
+					&& restPostBodyMediaType.equals(MediaType.APPLICATION_FORM_URLENCODED.toString())) { // FORM/POST
+				// PARAMETERS
+				if (StringUtils.isBlank(testKey) || StringUtils.isBlank(testValue)) {
+					throw new SubscriptionValidationException(
+							"Value & Key  in notificationMessage must have a values: " + notificationMessage);
+				}
+			} else {
+				if (notificationMessage.size() != 1) {
+					throw new SubscriptionValidationException(
+							"Only one array is allowed for notificationMessage when NOT using key/value pairs: "
+									+ notificationMessage);
+				} else if (testKey != null && !testKey.isEmpty()) {
+					throw new SubscriptionValidationException(
+							"Key in notificationMessage must be empty when NOT using key/value pairs: "
+									+ notificationMessage);
+				} else if (StringUtils.isBlank(testValue)) {
+					throw new SubscriptionValidationException(
+							"Value in notificationMessage must have a value when NOT using key/value pairs: "
+									+ notificationMessage);
+				}
+			}
+		}
+	}
+
+	/**
+	 * Validation of notificationMeta parameter Throws
+	 * SubscriptionValidationException if validation of the parameter fails due to
+	 * wrong format of parameter.
+	 *
+	 * @param notificationMeta
+	 */
+	private void validateNotificationMeta(String notificationMeta) throws SubscriptionValidationException {
+		String regex = ".*[\\s].*";
```

Comment:
See below comment for the naming of this regex variable.

Also, sometimes the checks are comparing with valid regex and sometimes with invalid regex like this one. It's a bit confusing to keep track of so is it possible to change so that every regex is negative or positive? It becomes more consistent and easier to follow along.

The checks are sometimes "if(!Pattern.matches(a,b))" and sometimes it's "if(Pattern.matches(a,b))".

Reply:
It is not part of my commit